src/snowflake/snowpark/mock/plan.py

```diff
@@ -1299,15 +1314,22 @@ def calculate_expression(
                         )
                     )
                 elif not ignore_nulls or offset == 0:
-                    res_cols.append(
-                        calculate_expression(
-                            window_function.expr,
-                            w.iloc[offset_idx],
-                            analyzer,
-                            expr_to_alias,
-                            keep_literal=True,
-                        )
+                    sub_window_res = calculate_expression(
+                        window_function.expr,
+                        w.iloc[offset_idx],
+                        analyzer,
+                        expr_to_alias,
+                        keep_literal=True,
                     )
+                    # we use the whole frame to calculate the type
+                    calculated_sf_type = calculate_expression(
```

Comment:
This could be moved outside the loop.

Reply:
We should raise an exception for coercion if the two sf_types are conflicting.

Reply:
updated to raise error if cur type != pre type